### PR TITLE
SD tutorial and benchmarks

### PR DESCRIPTION
**Pull request by jeffra**
* SD benchmark scripts
* SD tutorial showing how to deploy and evaluate a deployment
* Add MII package version that can be checked at runtime
* Fix GPRC max message size issue that I ran into when running multi-batch SD
* Added optional mii config to skip model validation checks (this currently takes >15sec and is not always needed)

### CODE REVIEW

File: examples/benchmark/txt2img/README.md
Comment (mrwyattii) [inline examples/benchmark/txt2img/README.md:8]: The second sentence here has a mistake: "Specifically, you will use both DeepSpeed Inference and DeepSpeed-MII to your deployment". Perhaps you meant "... to create your deployment"?

File: examples/benchmark/txt2img/README.md
Comment (mrwyattii) [inline examples/benchmark/txt2img/README.md:13]:
```suggestion
* [Environment and dependency setup](#environment-and-dependency-setup)
```

File: examples/benchmark/txt2img/README.md
Comment (mrwyattii) [inline examples/benchmark/txt2img/README.md:143]:
```suggestion
We've packaged up all that you need to deploy, query, and tear down an SD MII deployment in [mii-sd.py](mii-sd.py) which we will refer to going forward. You can run this example via:
```

File: mii/__init__.py
Comment (mrwyattii) [inline mii/__init__.py:15]: What is the purpose of this?

**Review by mrwyattii — APPROVED** [review]
LGTM, just a few minor suggestions to tutorial

File: mii/__init__.py
Comment (Thomas-MMJ) [reply mii/__init__.py:15]: elsewhere the author does

`assert version.parse(mii.__version__) >= version.parse("0.0.3"), "mii must be 0.0.3+"

so this provides a fake version number for versions prior to 0.0.3 that don't have one to allow the assert to work

**Comment by Suhail** [comment]
I'd recommend you mention how many steps are taken for the benchmark comparison. I didn't immediately find it so apologies if I missed it too.

**Comment by jeffra** [comment]
Thank you @Suhail for the feedback, I’ll update the tutorial before merging to add this info.

One thing for anyone reading this PR, we have pending changes coming later today on the DeepSpeed side related further performance updates and bug fixes. We’ll merge this PR once those our live.

**Pull request merged**
